### DISCUSSION TITLE
[bindings]How to make a binding take effect across different source file?

### DISCUSSION THREAD

**Discussion by huangweiwu** [wjakob/nanobind#293] [Q&A] [answered]
I bind a vector_float in a.cpp and a function in b.cpp has a vector<float> type argument.
When I call test_func(vector_float()) from python, it can't be auto castered to vector<float>.
```
// a.cpp
nb::bind_vector<std::vector<float>>(m, "vector_float");
```
```
// b.cpp
void test_func(vector<float> v) {};
```
```
// called from python
m.test_func(m.vector_float())
```

**Comment by wjakob** [comment, marked as answer]
Do you include `include/stl/vector.h` anywhere in your project?

**Reply by huangweiwu** [reply]
Yes you're right. I remove it and it's ok now.
